pkg/apis/options/logging.go

```diff
@@ -2,6 +2,7 @@ package options
 
 import (
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	requestutil "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests/util"
```

Comment:
The APIs code should import as little as possible from the rest of the package, to avoid the chance of circular dependencies. Should we move the const we are using to this package and import it into the requests util?

Reply:
Good point, I can bring it over (at least a copy) -- I only reused it here to keep the table default & helper default below internally consistent.

One special consideration, `requestutils` is a semi-special package in that its Get Header helpers are consumed by `logging` already (the key source of circular imports) -- which is the reason all those helpers are in their own package with limited other internal imports (just api middleware for scope).

Reply:
I'm tempted to leave this relying on `requestutil` (knowing that `logging` which is also imported has a sole internal import of `requestutil`) just so we keep everything aligned.

But super easy to just make these two spots set to `X-Request-Id` here explicitly for default setting to minimize the imports. I'll defer to you for the final call.